SPEC.md

```diff
@@ -693,7 +693,7 @@ JSON Schema for the Image Manifest (app image manifest, ACI manifest), conformin
     * **os**, **arch** can together be considered to describe the syscall ABI this image requires. **arch** is meaningful only if **os** is provided. If one or both values are not provided, the image is assumed to be OS- and/or architecture-independent. Currently supported combinations are listed in the [`types.ValidOSArch`](schema/types/labels.go) variable, which can be updated by an implementation that supports other combinations. The combinations whitelisted by default are (in format `os/arch`): `linux/amd64`, `linux/i386`, `freebsd/amd64`, `freebsd/i386`, `freebsd/arm`, `darwin/x86_64`, `darwin/i386`. See the [Operating System spec](OS-SPEC.md) for the environment apps can expect to run in given a known **os** label.
 * **app** (object, optional) if present, defines the default parameters that can be used to execute this image as an application.
     * **exec** (list of strings, required) executable to launch and any flags (must be non-empty; the executable must be an absolute path within the app rootfs; ACE can append or override)
-    * **user**, **group** (string, required) indicates either the UID/GID or the username/group name the app should run as (freeform string). If the user or group field begins with a "/", the owner and group of the file found at that absolute path inside the rootfs is used as the UID/GID of the process.
+    * **user**, **group** (string, required) indicates either the username/group name or the UID/GID the app should run as (freeform string). The user and group values may be all numbers to indicate a UID/GID, however it is possible on some systems (POSIX) to have usernames that are all numerical. The user and group values should first be resolved using the image's own `/etc/passwd` or `/etc/group`. If no valid matches are found, then if the string is all numerical, it should be converted to an integer and used as the UID/GID. If the user or group field begins with a "/", the owner and group of the file found at that absolute path inside the rootfs is used as the UID/GID of the process. Example values for the fields include `root`, `1000`, or `/usr/bin/ping`.
```

Comment:
Say I want to write a validator like http://yaml.travis-ci.org/ .  Should I have to upload the *entire* filesystem image for it to parse or explain to config to me?

Here's what I'd like such an example validator to tell me:

- if "uid" contains "/usr/bin/ping", it tells me: "the uid will be set according to the uid of this file on the filesystem!"
- if "uid" contains "username", it tells me: "the uid will be looked up from the username in /etc/passwd on the filesystem"
- if "uid" contains "=1000", it tells me: "the uid will be forcibly set to 1000"

If "uid" contains "1000", it tells me... what, exactly?  "the uid will either be forcibly set to 1000, or it will be looked up"?  I think we have a much stronger, simpler, more comprehensible system if we spec to eliminate "either" in a situation like this.

Ideally (if plain numerics as uids are retained for backcompat), I'd say a validator like this example should see "1000", tell you that it will have *either* one of two behaviors, and flag that yellow as ambiguous, and give you a recommendation to switch to the newer formats that are more explicit.